.github/workflows/sync_lab_release.yml

```diff
@@ -0,0 +1,71 @@
+name: Check for new JupyterLab releases
+
+on:
+  schedule:
+    - cron: 30 17 * * *
+  workflow_dispatch:
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  check_for_lab_updates:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Python
+        uses: actions/setup-python@v1
+        with:
+          python-version: '3.9'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install tbump
+
+      - name: Install Node
+        uses: actions/setup-node@v2
+        with:
+          node-version: '14.x'
+
+      - name: Install npm dependencies
+        run: |
+          npm install --global yarn
+          yarn install
+
+      - name: Check for new releases
+        shell: bash
+        run: |
+          set -eux
+          export LATEST=$(python scripts/get_latest_lab_version.py)
+          echo "latest=${LATEST}" >> $GITHUB_ENV
+          yarn set_jupyterlab_version ${LATEST}
+          if [[ ! -z "$(git status --porcelain package.json)" ]]; then
+            tbump --only-patch ${LATEST}-1 --non-interactive
+            yarn install
+          fi
+      - name: Create a PR if needed
+        shell: bash
+        env:
+          GITHUB_USER: ${{ secrets.GITHUB_USER }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        run: |
+          set -eux
+          # if resulted in any change:
```

Comment:
Wondering how this would work if we wanted to start using pre-release packages?

For example we might want to start using the upstream `4.0.0-rc.x` packages before the final release, and ship pre-releases of the desktop app. 

Looks like this script would try to downgrade to the stable release?

Reply:
We could condition the use of release/pre-release versions on branch fairly easily. That we are playing a catch up game here and still have a lot of legacy code so I don't know how soon we can start testing pre-releases.